### PR TITLE
Fix/configurable storage paths

### PR DESCRIPTION
**Pull request by AddressXception**
### Checklist
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] 🤔 **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] ✅ **DO** check open PR's to avoid duplicates.
- [x] ✅ **DO** keep pull requests small so they can be easily reviewed.
- [x] ✅ **DO** build locally before pushing.
- [x] ✅ **DO** make sure tests pass.
- [x] ✅ **DO** make sure any new changes are documented.
- [x] ✅ **DO** make sure not to introduce any compiler warnings.
- [x] ❌**AVOID** breaking the continuous integration build.
- [x] ❌**AVOID** making significant changes to the overall architecture.

### Description
This PR adds support for creating directories on paths if they do not already exist, when saving ballot and tally files out to the file system

💔Thank you!

### CODE REVIEW

File: src/electionguard/voting/encrypter.c [hunk 1]
```diff
@@ -182,6 +182,7 @@ Voting_Encrypter_encrypt_ballot(Voting_Encrypter encrypter,
 
     struct Voting_Encrypter_encrypt_ballot_r balotR;
     balotR.status = VOTING_ENCRYPTER_SUCCESS;
+    uint32_t selected_count;
```
Comment (pamtaro) [inline src/electionguard/voting/encrypter.c]: You'lll want to revert this file. The `selected_count` was removed in the latest merge in PR #49.

**Review by pamtaro — DISMISSED** [review]
Looks good 👍  just one file had some left-over changes and should be reverted

File: src/electionguard/directory.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
+
+
+#include <linux/limits.h>     /* PATH_MAX */
```
Comment (pamtaro) [inline src/electionguard/directory.c]: Build fails on Windows and Mac trying to include this header. 
Can we use `FILENAME_MAX` here from `stdio.h` instead? (that's also what we're using in other places where we're allocating a string for a path)

**Review by pamtaro — CHANGES_REQUESTED** [review]
unhappy build 😢

**Review by pamtaro — APPROVED** [review]
🎉

**Comment by echumley-msft** [comment]
LGTM alert resolved. Merging.

**Pull request merged**